### ISSUE TITLE
Hooked ES5 compatibility table for strict mode failures

### ISSUE THREAD

**Issue by t2ym** [t2ym/thin-hook#112]
## Status (as of 0.0.150)
- [x] reserved words
- [x] "this" is undefined in functions
- [x] legacy octal is a SyntaxError
- [x] assignment to unresolvable identifiers is a ReferenceError
- [x] assignment to eval or arguments is a SyntaxError
- [x] assignment to non-writable properties is a TypeError
- [x] eval or arguments bindings is a SyntaxError
- [x] arguments.callee is a TypeError
- [x] eval() can't create bindings
- [x] deleting bindings is a SyntaxError
- [x] deleting non-configurable properties is a TypeError
- [x] "with" is a SyntaxError
- [x] repeated parameter names is a SyntaxError

## Hooked ES5 compatibility table for strict mode failures as of 0.0.114

<img src="https://raw.githubusercontent.com/wiki/t2ym/thin-hook/hooked_strict_mode.png" width="500px">

**Comment by t2ym** [comment]
There are the very same failures on Firefox 55 as well.

**Comment by t2ym** [comment]
The failures do not simply mean the features are unavailable.  There may be some other root causes of the failures in the test for the compatibility table generation, e.g., `Function` constructor compatibility.

**Comment by t2ym** [comment]
[This fix](https://github.com/t2ym/thin-hook/commit/eb4fcf6195d616ddd9cd35695a8ac5df15977f04) for Issue #111 is effective for 
- "this" is undefined for functions
- arguments.callee is a TypeError

**Comment by t2ym** [comment]
## Root Cause:
The wrapper function `(script, eval) => eval(script)` violates the strict mode restrictions.

```javascript
var df = __hook__(Function, null, [
"'use strict';try { eval('function f(x, x) { }'); } catch (err) { return err instanceof SyntaxError; }"], 'context', true)
Uncaught SyntaxError: Unexpected eval or arguments in strict mode
    at new Function (<anonymous>)
```
- hooked version (violating the strict mode restrictions)
```javascript
  'use strict';
  return __hook__(() => {
    try {
      __hook__('()', hook, [
        'eval',
        [
          '__hook__',
          [[
              'f',
              {}
            ]],
          'method',
          { arguments: true }
        ]
      ], 'f')('function f(x, x) { }', (script, eval) => eval(script)); // this wrapper function (script, eval) => eval(script) violates the strict mode restrictions
    } catch (err) {
      return err instanceof hook.global(__hook__, 'f', 'SyntaxError', 'get')._p_SyntaxError;
    }
  }, null, arguments, 'f');
```

**Comment by t2ym** [comment]
## Other Root Causes:
- `'use strict';` is missing in the script

**Comment by t2ym** [comment]
Fixed as of 0.0.150
